drivers/battery/Kconfig

```diff
@@ -0,0 +1,46 @@
+# Copyright 2022 Google LLC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig BATTERY
+	bool "Battery drivers"
+	depends on I2C
+	help
+	  Enable battery driver configuration.
+
+if BATTERY
+
+config BATTERY_SHIP_MODE_WRITES
```

Comment:
This could be moved into the devicetree.  For systems that support detecting the installed battery at runtime, the number of writes needed for ship mode could vary between battery types.  This would require changing `ship_mode_info` to include the write count as determined from the devicetree.